iothub/device/src/IotHubBaseClient.cs

```diff
@@ -308,6 +339,18 @@ public async Task SendTelemetryAsync(IEnumerable<TelemetryMessage> messages, Can
         /// <exception cref="OperationCanceledException">The operation has been canceled.</exception>
         /// <exception cref="IotHubClientException">An error occured when communicating with IoT hub service.</exception>
         /// <exception cref="ObjectDisposedException">The client has been disposed.</exception>
+        /// <example>
+        /// <code language="csharp">
+        /// await client.SetDirectMethodCallbackAsync(OnDirectMethodCalledAsync, cancellationToken);
+        /// //...
+        /// 
+        /// Task<DirectMethodResponse/> OnDirectMethodCalledAsync(DirectMethodRequest directMethodRequest)
+        /// {
+        ///     // Add method request handling logic as needed
```

Comment:
Let's show inspecting the method name to compare to a known value, and then maybe getting the expected payload type.